# 인공지능수학: 통계학
# 숙제 13 (HW 13)


이름: 신용주

학번: K2021022

# 아래 문제를 해결하는 코드를 작성하고, 그 결과를 쓰시오.

1. 어떤 농장에서 생산된 계란 35 개의 표본을
뽑아서 아래 변수 w에 저장하였다. 계란의 평균 무게에 대한 95% 신뢰 구간을
구하시오.

In [ ]:
w=[22,
 24,
 23,
 22,
 21,
 27,
 26,
 27,
 25,
 29,
 21,
 25,
 28,
 26,
 20,
 21,
 25,
 20,
 28,
 23,
 26,
 20,
 21,
 27,
 24,
 28,
 25,
 26,
 20,
 24]

In [ ]:
# 여기에 코드를 작성하시오.
import numpy as np
xbar=np.mean(w)
sd=np.std(w, ddof=1)
print("평균 %.2f,표준편차:%.2f"%(xbar,sd))
import scipy.stats
alpha=0.05
zalpha=scipy.stats.norm.ppf(1-alpha/2)
print("zalpha:",zalpha)
print((24.13-0.475 , 24.13+0.475))

평균 24.13,표준편차:2.81
zalpha: 1.959963984540054
(23.654999999999998, 24.605)


(23.655 , 24.605)

2. 어떤 정책에 대한 찬성율을 조사하기 위해 783명을
랜덤하게 선택하여 찬성 여부를 조사하였는데, 이 중 475 명이 찬성하였다. 찬성율 𝑝의 95% 신뢰구간 (confidence interval) 을 구하시오.

In [ ]:
# 2.번 문제에 대한 코드
import numpy as np
import scipy.stats
x=475
n=783
phat = x / n 
alpha=0.05
xalpha = scipy.stats.norm.ppf(1-alpha/2)
sd=np.sqrt(phat*(1-phat)/n)
print("phat%.3f,zalpha:%.3f,sd:%.3f"%(phat,zalpha,sd))
ci=[phat-zalpha*sd,phat+zalpha*sd]
print(ci)


phat0.607,zalpha:1.960,sd:0.017
[0.5724252520070054, 0.640856995758001]


(0.572 , 0.640 )

3. 위 1번의 문제에서 계란의 평균 무게는 23.5 그램으로 알려져 있다 . 새로운 사료를 도입한 후에 생산된 계란 35 개의 무게가 위 1번 문제의 w 변수에 저장된 것이라면, 새로운 사료가 평균적으로 더 무거운 계란을 생산한다고 할 수 있는가? 유의수준 5%로 검정하시오.

In [ ]:
# 코드 작성
import numpy as np 
w=[22,24,23,22,21,27,26,27,25,29,21,25,28,26,20,21,25,20,28,23,26,20,21,27,24,28,25,26,20,24]
mu=23.5
xbar=np.mean(w)
sd=np.std(w, ddof=1)
print("평균%.2f, 표준편차:%.2f"%(xbar, sd))
z=(xbar-mu)/(sd/np.sqrt(len(w)))
print("검정통계량:",z)
alpha=0.05
import scipy.stats
cri = scipy.stats.norm.ppf(1-alpha/2)
print("임계값:",cri)

평균24.13, 표준편차:2.81
검정통계량: 1.233196175191864
임계값: 1.959963984540054


귀무가설을 기각할 수 없다.